File: implementation/src/benchmark/kotlin/com/aoc/intcode/network/IntCodeComputerNetworkTest.java
package com.aoc.intcode.network;

import com.aoc.AdventBenchmarkingSuite;
import com.aoc.BenchmarkInputReader;
import com.aoc.TestInputReader;
import com.aoc.input.Day;
import org.openjdk.jmh.annotations.Benchmark;
import org.openjdk.jmh.annotations.Warmup;

@AdventBenchmarkingSuite
public class IntCodeComputerNetworkTest {

    private static final String software = BenchmarkInputReader.read(Day.Companion.from(23)).asSingleString();

    @Benchmark
    @Warmup(iterations = 1)
    public void solutionPartOne() {
        final IntCodeComputerNetwork network = new IntCodeComputerNetwork(software);
        network.attachPacketAnalyserAtTargetAddress(new NetworkAddress(255));
        network.boot();
    }

    @Benchmark
    public void solutionPartTwo() {
        new IntCodeComputerNetwork(software).boot();
    }
}

File: implementation/src/test/kotlin/com/aoc/intcode/network/IntCodeComputerNetworkTest.kt
package com.aoc.intcode.network

import assertk.assertThat
import assertk.assertions.isEqualTo
import com.aoc.Day
import com.aoc.input.InputReader
import org.junit.jupiter.api.Test

class IntCodeComputerNetworkTest {

    private val software = InputReader.read<String>(Day(23)).asSingleString()

    @Test
    fun solutionPartOne() {
        val network = IntCodeComputerNetwork(software)
        network.attachPacketAnalyserAtTargetAddress(NetworkAddress(255))
        val packet = network.boot()
        assertThat(packet.data.y).isEqualTo(23815)
    }

    @Test
    fun solutionPartTwo() {
        val network = IntCodeComputerNetwork(software)
        val packet = network.boot()
        assertThat(packet.data.y).isEqualTo(16666)
    }
}